src/ray/object_manager/object_manager.cc

```diff
@@ -47,7 +47,7 @@ ObjectManager::ObjectManager(asio::io_service &main_service,
       connection_pool_() {
   RAY_CHECK(config_.max_sends > 0);
   RAY_CHECK(config_.max_receives > 0);
-  RAY_CHECK(config_.max_push_retries > 0);
+  RAY_CHECK(config_.push_timeout_ms > 0);
```

Comment:
I think we can allow this value to be <= 0.
if <0, wait indefinitely;
if ==0, do not wait, fail directly.
Does this make more sense?